src/python/gem5/simulate/exit_event_generators.py

```diff
@@ -95,3 +98,29 @@ def default_workend_generator():
     while True:
         m5.stats.dump()
         yield False
+
+
+def default_simpoint_generator():
+    """
+    A default generator for simpoints. It will do nothing.
+    """
+    defaultBehaviorWarning(
+        "default_simpoint_generator",
+        "A default generator for simpoints. It will do nothing.")
+    while True:
+        yield False
+
+def simpoint_dump_reset_generator():
+    while True:
+        # only print for testing purpose
+        print("got to the simpoint_dump_reset_gnereator\n")
+        m5.stats.dump()
+        m5.stats.reset()
+        yield False
+
+def simpoint_save_checkpoint_generator(checkpoint_dir: Path):
+    while True:
+        # only print for testing purpose
+        print("got to the simpoint_save_checkpoint_generator\n")
+        m5.checkpoint(joinpath(checkpoint_dir,f"cpt.{str(m5.curTick())}"))
```

Comment:
We should think more about how to do this. I don't love that we're using the `m5` interface here. @BobbyRBruce, I think that we need to have a way to get to the simulator instance from these generators.